http-client/src/main/java/org/camunda/connect/httpclient/impl/RequestConfigOption.java

```diff
@@ -0,0 +1,148 @@
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setContentCompressionEnabled((boolean) value);
+    }
+  },
+  COOKIE_SPEC("cookie-spec") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setCookieSpec((String) value);
+    }
+  },
+  DECOMPRESSION_ENABLED("decompression-enabled") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setDecompressionEnabled((boolean) value);
+    }
+  },
+  EXPECT_CONTINUE_ENABLED("expect-continue-enabled") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setExpectContinueEnabled((boolean) value);
+    }
+  },
+  LOCAL_ADDRESS("local-address") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setLocalAddress((InetAddress) value);
+    }
+  },
+  MAX_REDIRECTS("max-redirects") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setMaxRedirects((int) value);
+    }
+  },
+  NORMALIZE_URI("normalize-uri") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setNormalizeUri((boolean) value);
+    }
+  },
+  PROXY("proxy") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setProxy((HttpHost) value);
+    }
+  },
+  PROXY_PREFERRED_AUTH_SCHEMES("proxy-preferred-auth-scheme") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setProxyPreferredAuthSchemes((Collection<String>) value);
+    }
+  },
+  REDIRECTS_ENABLED("relative-redirects-allowed") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setRedirectsEnabled((boolean) value);
+    }
+  },
+  RELATIVE_REDIRECTS_ALLOWED("relative-redirects-allowed") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setRelativeRedirectsAllowed((boolean) value);
+    }
+  },
+  SOCKET_TIMEOUT("socket-timeout") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setSocketTimeout((int) value);
+    }
+  },
+  STALE_CONNECTION_CHECK_ENABLED("stale-connection-check-enabled") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setStaleConnectionCheckEnabled((boolean) value);
+    }
+  },
+  TARGET_PREFERRED_AUTH_SCHEMES("target-preferred-auth-schemes") {
+    @Override
+    public void apply(Builder configBuilder, Object value) {
+      configBuilder.setTargetPreferredAuthSchemes((Collection<String>) value);
+    }
+  };
+
+  private String name;
+
+  private RequestConfigOption(String name) {
+    this.name = name;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void apply(Builder configBuilder, Object value) {
+  }
```

Comment:
Should this be `abstract` so every enum instance has to implement it? Or would you rather have it like this and then a new option does not have to implement it and do nothing by default?